src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,90 @@
+package frc.trigon.robot.motorsimulation;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import frc.trigon.robot.utilities.Conversions;
+
+public abstract class MotorSimulation {
```

Comment:
Add a javadoc to explain this class
Make it so when you're sending a pid request, you'll only need to call it once to reach a target position - the pid should continuously run using a notifier. We need this so the motor could act as similarly as possible to the real motor, as this is what the real motor does.
i.e. I can use `motor.setControl(motionMagicVoltage.withPosition(0.5)` once, and the motor will keep on trying to reach 0.5 without me needing to use setControl again.